modules/core/src/test/scala/sangria/util/CatsSupport.scala

```diff
@@ -215,26 +215,23 @@ object CatsScenarioExecutor extends FutureResultSupport {
 
   def executeAction(given: Given[Any, Any], action: Action) = action match {
     case Parse =>
-      import sangria.parser.DeliveryScheme.Either
+      ParsingResult(
+        QueryParser.parse(given.query).toEither.left.map(_.asInstanceOf[SangriaSyntaxError]))
 
-      ParsingResult(QueryParser.parse(given.query).left.map(_.asInstanceOf[SangriaSyntaxError]))
     case Validate(rules) =>
-      import sangria.parser.DeliveryScheme.Throw
-
       ValidationResult(
         new RuleBasedQueryValidator(rules.toList)
-          .validateQuery(given.schema, QueryParser.parse(given.query)))
-    case Execute(validate, value, vars, op) =>
-      import sangria.parser.DeliveryScheme.Throw
+          .validateQuery(given.schema, QueryParser.parse(given.query).get))
 
+    case Execute(validate, value, vars, op) =>
       val validator = if (validate) QueryValidator.default else QueryValidator.empty
 
       ExecutionResult(
         Try(
           Executor
             .execute(
               given.schema,
-              QueryParser.parse(given.query),
+              QueryParser.parse(given.query).get,
```

Comment:
Instead of calling `.get`, we can `map` on it?
```
QueryParser.parse(given.query).map(ast => Executor.execute(given.schema, ast,...))
```

Reply:
One could `map` on it, but that would change the behavior of the code. The `import sangria.parser.DeliveryScheme.Throw` above it says that `QueryParser.parse` should throw an exception on failure.

Reply:
This exception would be catch by the `Try` above.

Reply:
I see. You're proposing to replace lines 230-240 in the new code. You'd need the `.await` as well. Sure, one could do that. My goal wasn't to analyze the code to make it aesthetic, just trying to get the PR done.